### DISCUSSION TITLE
How to use ThemeResource from code?

### DISCUSSION THREAD

**Discussion by StevoSM** [microsoft/microsoft-ui-xaml#10001] [Q&A]
In XAML, I can style a Line with `Stroke="{ThemeResource SystemControlForegroundBaseMediumLowBrush}"`.  If I am composing a Line in C++ code, how would I accomplish the same thing?  

As in `line.Stroke(-- what goes here? --);`

**Comment by castorix** [comment]
You can read _Application.Current.Resources_
  
  In C# :
  
```
  var brush = (Brush)Application.Current.Resources["SystemControlForegroundBaseMediumLowBrush"];
  line.Stroke = brush;
```
  
  I cannot test in C++, but if I ask the translation to ChatGPT, it gives me (not always 100% correct...) :
  
```
  auto brush = unbox_value<Brush>(Application::Current().Resources().Lookup(winrt::box_value(L"SystemControlForegroundBaseMediumLowBrush")));
  line.Stroke(brush);
```

**Reply by Zakariathr22** [reply]
I think that the provided solution uses `StaticResource`, not `ThemeResource`. This means it won't dynamically respond to theme changes (like switching between light and dark mode). 

I raised a proposal in the [Microsoft UI XAML GitHub repository](https://github.com/microsoft/microsoft-ui-xaml/issues/9464) addressing the need for direct color changes using `ThemeResource` in C# for WinUI3. In the meantime, I’ve found a **workaround** that works in C# by leveraging `Border` controls to hold theme brushes and applying them in the `Loaded` and `ActualThemeChanged` events (you can find the workaround [here](https://github.com/microsoft/microsoft-ui-xaml/issues/9464#issuecomment-2258432746)).

*Notes:* 
*- I am not sure if this is the best practice for C#, but at least it is doing the job.*
*- The workaround still needs a translation to C++.*

**Reply by NicholasChrzan** [reply]
Can't you just create a style in your resource dictionary that uses theme resources then apply that in code behind?

**Reply by Zakariathr22** [reply]
@NicholasChrzan
Yes, this is actually an easier way, I don't know how I didn't think of it before 😂. 
[I edited the workaround](https://github.com/microsoft/microsoft-ui-xaml/issues/9464#issuecomment-2258432746).
Thanks for your help!

**Reply by StevoSM** [reply]
Thanks everyone for your contributions.  Given the above, I ended up with a solution that leaves the stroke defined in XAML and loaded that as a custom control.  The actual controls (implemented in more than one control) are more than just the Line, and ended up being a more direct and efficient solution than doing much of it in code as I initially planned.
